test/Microsoft.DotNet.Tools.Tests.Utilities/Commands/DotnetCommand.cs

```diff
@@ -15,6 +17,41 @@ public DotnetCommand()
         public DotnetCommand(string dotnetUnderTest)
             : base(dotnetUnderTest)
         {
+            // Some scripts (e.g. RunCsc/RunVbc) depend on the correct `dotnet`
+            // being in the PATH. When tests use `DotnetUnderTest.WithBackwardsCompatibleRuntimes`,
+            // this means that the up-to-date dotnet command is resolved by the
+            // old-runtime-targeting RunCsc/Roslyn, causing compatability issues.
+            // So, detect if we're running a dotnet that's not just a plain "dotnet",
+            // and if so, make sure it's on the PATH.
+            var containingDirectory = Path.GetDirectoryName(dotnetUnderTest);
+            if (!string.IsNullOrEmpty(containingDirectory))
+            {
+                // if it's already set, use that
+                string path;
+                if (!Environment.TryGetValue("PATH", out path))
+                {
+                    // otherwise grab it from the environment
```

Comment:
@khyperia  

> as they explain the why, instead of just the how

I disagree, C# is a high level language that is capable explain why in the code. But I know it is very debatable 

>  And for the feedback that understanding comments is required before changes to the code can be made

If it is this case, could you add test coverage for this? Comment will out of sync and people make mistake. I hope at least test will fail due to it and point me to the code. (Even it is a test helper method). I don't want this code to be another legacy part of the code I am afraid to change.